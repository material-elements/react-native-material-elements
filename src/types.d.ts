export * from './components/types';
export * from './types/index';
