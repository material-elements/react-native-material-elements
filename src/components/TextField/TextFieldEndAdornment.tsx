import React from 'react';
import { ActivityIndicatorProps } from 'react-native';
import { ActivityIndicator } from '../ActivityIndicator';
import { Box } from '../Box';
import { BoxProps } from '../types';

export interface TextFieldEndAdornmentProps {
  loading?: boolean;
  endAdornment?: React.ReactNode;
  showLoadingIndicatorWhenFocused?: boolean;
  loadingIndicatorProps?: ActivityIndicatorProps;
  isFocused?: boolean;
  endAdornmentContainerProps?: Omit<BoxProps, 'children'>;
}

const TextFieldEndAdornment: React.FC<TextFieldEndAdornmentProps> = ({
  loading,
  endAdornment,
  showLoadingIndicatorWhenFocused,
  isFocused,
  loadingIndicatorProps,
  endAdornmentContainerProps,
}) => {
  if (!loading && !endAdornment) {
    return null;
  }

  let element: React.ReactNode;

  if (endAdornment) {
    element = endAdornment;
  } else {
    if (showLoadingIndicatorWhenFocused && isFocused && loading) {
      element = <ActivityIndicator {...loadingIndicatorProps} />;
    } else if (!showLoadingIndicatorWhenFocused && loading) {
      element = <ActivityIndicator {...loadingIndicatorProps} />;
    }
  }

  if (!element) {
    return null;
  }

  return (
    <Box sx={{ me: 8, ms: 8 }} {...endAdornmentContainerProps}>
      {element}
    </Box>
  );
};

export default TextFieldEndAdornment;
