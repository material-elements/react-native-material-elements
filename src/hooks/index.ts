export * from './useRestyle';
export * from './useThemedProps';
