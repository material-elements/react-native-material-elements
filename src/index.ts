export * from './components';
export * from './hooks';
export * from './libraries';
export * from './utils';
