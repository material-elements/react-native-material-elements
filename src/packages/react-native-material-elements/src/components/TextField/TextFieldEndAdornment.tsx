import React from 'react';
import { ActivityIndicatorProps } from 'react-native';
import { ActivityIndicator } from '../ActivityIndicator';
import { Box } from '../Box';
import { BoxProps } from '../types';

export interface TextFieldEndAdornmentProps {
  loading?: boolean;
  endAdornment?: React.ReactNode;
  showLoadingIndicatorWhenFocused?: boolean;
  loadingIndicatorProps?: ActivityIndicatorProps;
  isFocused?: boolean;
  endAdornmentContainerProps?: Omit<BoxProps, 'children'>;
  testID?: string;
}

const TextFieldEndAdornment: React.FC<TextFieldEndAdornmentProps> = ({
  loading,
  endAdornment,
  showLoadingIndicatorWhenFocused,
  isFocused,
  loadingIndicatorProps,
  endAdornmentContainerProps,
  testID,
}) => {
  if (!loading && !endAdornment) {
    return null;
  }

  let element: React.ReactNode = null;

  if (endAdornment) {
    element = endAdornment;
  } else if ((showLoadingIndicatorWhenFocused && isFocused && loading) || (!showLoadingIndicatorWhenFocused && loading)) {
    element = <ActivityIndicator {...loadingIndicatorProps} />;
  }

  if (!element) {
    return null;
  }

  return (
    <Box sx={{ me: 8, ms: 8 }} {...endAdornmentContainerProps} testID={testID}>
      {element}
    </Box>
  );
};

export default TextFieldEndAdornment;
