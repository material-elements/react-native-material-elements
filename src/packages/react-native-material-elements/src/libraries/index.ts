export * from './themes';
export * from './style';
