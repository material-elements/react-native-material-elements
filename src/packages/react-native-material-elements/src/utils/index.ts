export * from './styleGenerator';
export * from './utils';
export * from './dimensions';
