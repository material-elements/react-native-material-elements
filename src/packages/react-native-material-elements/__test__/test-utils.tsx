import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';
import { ThemeProviderProps } from '../src/libraries/types';

interface CustomRenderOptions extends Omit<RenderOptions, 'wrapper'> {
  themeProps?: Partial<ThemeProviderProps>;
}

export const ThemeWrapper: React.FC<ThemeProviderProps> = ({ children, ...props }) => {
  return <ThemeProvider {...props}>{children}</ThemeProvider>;
};

export const customRender = (ui: React.ReactElement, options?: CustomRenderOptions) => {
  const { themeProps, ...rest } = options || {};
  return render(ui, {
    wrapper: wrapperProps => <ThemeWrapper {...themeProps}>{wrapperProps.children}</ThemeWrapper>,
    ...rest,
  });
};

export * from '@testing-library/react-native';
export { customRender as render };
