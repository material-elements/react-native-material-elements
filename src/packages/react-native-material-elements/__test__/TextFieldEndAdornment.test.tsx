import React from 'react';
import { ActivityIndicator, Text } from '../src';
import TextFieldEndAdornment from '../src/components/TextField/TextFieldEndAdornment';
import { render } from './test-utils';

describe('TextFieldEndAdornment', () => {
  const mockTextFieldEndAdornmentTestId = 'text-field-end-adornment-test-id';
  const mockLoadingIndicatorTestId = 'loading-indicator-test-id';

  const mockElementText = 'Mock Element';
  const mockElement = <Text>{mockElementText}</Text>;

  it('should render component correctly', () => {
    const { toJSON } = render(<TextFieldEndAdornment />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render the adornment component', () => {
    const { getByText } = render(<TextFieldEndAdornment testID={mockTextFieldEndAdornmentTestId} endAdornment={mockElement} />);

    const element = getByText(mockElementText);

    expect(element).toBeDefined();
  });

  it('should render the loading indicator when loading prop passed', () => {
    const { getByTestId } = render(
      <TextFieldEndAdornment loading loadingIndicatorProps={{ testID: mockLoadingIndicatorTestId }} />,
    );

    const loaderComponent = getByTestId(mockLoadingIndicatorTestId);
    expect(loaderComponent).toBeDefined();
  });

  it('should render the loading indicator when loading prop passed', () => {
    const { getByTestId } = render(<TextFieldEndAdornment loadingIndicatorProps={{ testID: 'loader' }} loading />);

    const loader = getByTestId('loader');
    expect(loader).toBeDefined();
  });

  it('renders ActivityIndicator when loading=true and showLoadingIndicatorWhenFocused=false', () => {
    const { getByTestId, UNSAFE_getByType } = render(
      <TextFieldEndAdornment loading showLoadingIndicatorWhenFocused={false} testID="endAdornment" />,
    );

    const container = getByTestId('endAdornment');
    expect(container).toBeTruthy();
    expect(UNSAFE_getByType(ActivityIndicator)).toBeTruthy();
  });

  it('renders ActivityIndicator when loading=true, showLoadingIndicatorWhenFocused=true, and isFocused=true', () => {
    const { getByTestId, UNSAFE_getByType } = render(
      <TextFieldEndAdornment loading showLoadingIndicatorWhenFocused isFocused testID="endAdornment" />,
    );

    const container = getByTestId('endAdornment');
    expect(container).toBeTruthy();
    expect(UNSAFE_getByType(ActivityIndicator)).toBeTruthy();
  });

  it('returns null when no loading and no endAdornment', () => {
    const { toJSON } = render(<TextFieldEndAdornment />);
    expect(toJSON()).toBeNull();
  });

  it('returns null when showLoadingIndicatorWhenFocused=true but not focused', () => {
    const { toJSON } = render(<TextFieldEndAdornment loading showLoadingIndicatorWhenFocused isFocused={false} />);
    expect(toJSON()).toBeNull();
  });
});
