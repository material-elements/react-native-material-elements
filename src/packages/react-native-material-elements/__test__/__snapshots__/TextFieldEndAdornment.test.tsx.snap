// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldEndAdornment should render component correctly 1`] = `null`;
